test/function.localstack.test.ts

```diff
@@ -0,0 +1,332 @@
+    },
+    async (context) => {
+      await testFunction(context.function, {}, 5);
+    }
+  );
+
+  testResource(
+    "Call Lambda payload",
+    (parent) => {
+      const func = new Function(
+        parent,
+        "function",
+        async (event: { val: string }) => {
+          return `value: ${event.val}`;
+        }
+      );
+
+      return { outputs: { function: func.resource.functionName } };
+    },
+    async (context) => {
+      await testFunction(context.function, { val: "hi" }, "value: hi");
+    }
+  );
+
+  testResource(
+    "Call Lambda throw error",
+    (parent) => {
+      const func = new Function(parent, "function", async () => {
+        throw Error("AHHHHHHHHH");
+      });
+
+      return {
+        outputs: {
+          function: func.resource.functionName,
+        },
+      };
+    },
+    async (context) => {
+      await testFunction(
+        context.function,
+        {},
+        { errorMessage: "AHHHHHHHHH", errorType: "Error" }
+      );
+    }
+  );
+
+  testResource(
+    "Call Lambda put event to bus",
+    (parent) => {
+      const bus = new EventBus(parent, "bus");
+      const busbus = new aws_events.EventBus(parent, "busbus");
+      const func = new Function(parent, "function", async () => {
+        return `${bus.eventBusArn} ${busbus.eventBusArn}`;
+      });
+
+      return {
+        outputs: {
+          bus: bus.eventBusArn,
+          busbus: busbus.eventBusArn,
+          function: func.resource.functionName,
+        },
+      };
+    },
+    async (context) => {
+      await testFunction(
+        context.function,
+        {},
+        `${context.bus} ${context.busbus}`
+      );
+    }
+  );
+
+  testResource(
+    "Call Lambda AWS SDK put event to bus with reference",
+    (parent) => {
+      const bus = new EventBus<any>(parent, "bus");
+
+      // Necessary to keep the bundle small and stop the test from failing.
+      // See https://github.com/sam-goodwin/functionless/pull/103#issuecomment-1116396779
+      const putEvents = $AWS.EventBridge.putEvents;
+      const func = new Function(parent, "function", async () => {
+        const result = putEvents({
+          Entries: [
+            {
+              EventBusName: bus.eventBusArn,
+              Source: "MyEvent",
+              DetailType: "DetailType",
+              Detail: "{}",
+            },
+          ],
+        });
+        return result.FailedEntryCount;
+      });
+
+      return { outputs: { function: func.resource.functionName } };
+    },
+    async (context) => {
+      await testFunction(context.function, {}, 0);
+    }
+  );
+
+  // See https://github.com/sam-goodwin/functionless/pull/103#issuecomment-1116396779
+  testResource.skip(
+    "Call Lambda AWS SDK put event to bus without reference",
+    (parent) => {
+      const bus = new EventBus<EventBusRuleInput>(parent, "bus");
+
+      const func = new Function(parent, "function", async () => {
+        const result = $AWS.EventBridge.putEvents({
```

Comment:
Is this really not supported?

Reply:
It is supported, but it breaks in jest.

Essentially, pulumi serializes the entire `aws.js` file, which has `jest`'s injection. Then `global[ts-jest]` doesn't exist and fails in the lambda when its run on localstack.

Doing this outside of tests works, but does bloat the output bundle. #122 is supposed to reduce the change of this case.